Jdp230502 15

WorstWorm opened the pull request:
Implementation of User entity with UserRopository, and UserMapper class; small corrections in UserController and build.gradle

WorstWorm committed:
JDP230502-15 - Implementation of user entity with UserRepository and UserMapper; small corrects in UserController and needed modifications in build.gradle

 On branch JDP230502-15
 Changes to be committed:
	modified:   build.gradle
	modified:   src/main/java/com/kodilla/ecommercee/controller/UserController.java
	new file:   src/main/java/com/kodilla/ecommercee/domain/User.java
	new file:   src/main/java/com/kodilla/ecommercee/mapper/UserMapper.java
	new file:   src/main/java/com/kodilla/ecommercee/repository/UserRepository.java

WorstWorm committed:
JDP230502-15 - corrections according to errors pointed out by Circle

 On branch JDP230502-15
 Your branch is up to date with 'origin/JDP230502-15'.

 Changes to be committed:
	modified:   build.gradle
	modified:   src/main/java/com/kodilla/ecommercee/dto/UserDto.java
	modified:   src/main/java/com/kodilla/ecommercee/mapper/UserMapper.java

WorstWorm committed:
JDP230502-15 more corrections according to errors pointed out by Circle

 On branch JDP230502-15
 Your branch is up to date with 'origin/JDP230502-15'.

 Changes to be committed:
	modified:   build.gradle

WorstWorm committed:
JDP230502-15 build.gradle restored to its initial state

 On branch JDP230502-15
 Your branch is up to date with 'origin/JDP230502-15'.

 Changes to be committed:
	modified:   build.gradle

sew-k commented on src/main/java/com/kodilla/ecommercee/domain/User.java:
```diff
@@ -0,0 +1,32 @@
+    @NotNull
+    @GeneratedValue
+    @Column(name = "id", unique = true)
+    private long id;
```
shouldn't we use Long wrapper like in GenericEntity.class ?

sew-k commented on src/main/java/com/kodilla/ecommercee/repository/UserRepository.java:
```diff
@@ -0,0 +1,19 @@
+
+@Repository
+public interface UserRepository extends CrudRepository<User, Long> {
+    List<User> findAll();
```
I'm not sure if it's necessary, but maybe is it better to use Override annotation?
edit: over all this class fields

JakubFir replied:
I suggest useing JPARepository instead of crud like in GenericEntity so we dont have to implement standard crud methods

WorstWorm replied:
> I suggest useing JPARepository instead of crud like in GenericEntity so we dont have to implement standard crud methods

I don't quite understand. As far as I can see, JpaRepository extends CrudRepository - it has a few additional functionalities (mainly related to sorting), but I don't know if we need it in any way, and that's pretty much of it (if I'm wrong - let me know).
I think this is one of those things we have to decide together, but in general I think it's better to use CrudRepository. One of many comparisons:
https://www.tutorialspoint.com/difference-between-crudrepository-and-jparepository-in-java

annapawlas replied:
I think we can use CrudRepository like we did with the Task project.

sew-k replied:
I also use CrudRepository like in past projects.

WorstWorm replied:
I have two CR here already (thx annapawlas and sew-k) and I could merge, but for now we have 3 votes for Crud, 1 for Jpa and two people with no comment.
I'll wait for one more support for Crud (or more details from JakubFir - maybe there is a good reason to use Jpa) before merging.

WorstWorm replied:
Bartosz confirmed Crud so I'm closing the discussion and merging. Thx for your input People...

WorstWorm committed:
Changes suggested by sew-k (long to Long on User and @Override annotations in UserRepository).

 On branch JDP230502-15
 Your branch is up to date with 'origin/JDP230502-15'.

 Changes to be committed:
	modified:   src/main/java/com/kodilla/ecommercee/domain/User.java
	modified:   src/main/java/com/kodilla/ecommercee/repository/UserRepository.java

WorstWorm committed:
Merge branch 'master' into JDP230502-15
